addons/mail/static/tests/discuss_app/discuss_tests.js

```diff
@@ -1836,3 +1836,68 @@ QUnit.test("Escape key should focus the composer if it's not focused", async ()
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "Mario Party",
+        channel_type: "channel",
+    });
+    const { openDiscuss } = await start();
+    openDiscuss(channelId);
+    await click(".o-mail-Discuss-header button[title='Notification Settings']");
+    await contains(".o-discuss-NotificationSettings button", { text: "All Messages" });
+    await contains(".o-discuss-NotificationSettings button", { text: "Mentions Only" });
+    await contains(".o-discuss-NotificationSettings button", { text: "Nothing" });
+
+    await click(".o-discuss-NotificationSettings button[title='Mute Channel']");
+    await contains(".dropdown-item[title='For 15 minutes']");
+    await contains(".dropdown-item[title='For 1 hour']");
+    await contains(".dropdown-item[title='For 3 hours']");
+    await contains(".dropdown-item[title='For 8 hours']");
+    await contains(".dropdown-item[title='For 24 hours']");
+    await contains(".dropdown-item[title='Until I turn it back on']");
+});
+
+QUnit.test("Notification settings: mute channel will change the style of sidebar", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "Mario Party",
+        channel_type: "channel",
+    });
+    const { openDiscuss } = await start();
+    openDiscuss(channelId);
+
+    await contains(".o-mail-DiscussSidebar-item", { text: "Mario Party" });
+    await contains(".o-mail-DiscussSidebar-item[class*='opacity-50']", {
+        text: "Mario Party",
+        count: 0,
+    });
+
+    await click(".o-mail-Discuss-header button[title='Notification Settings']");
+    await click(".o-discuss-NotificationSettings button[title='Mute Channel']");
+    await click(".dropdown-item[title='For 15 minutes']");
+
+    await contains(".o-mail-DiscussSidebar-item", { text: "Mario Party" });
+    await contains(".o-mail-DiscussSidebar-item[class*='opacity-50']", { text: "Mario Party" });
+});
+
+QUnit.test("Notification settings: mute/unmute channel works correctly", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "Mario Party",
+        channel_type: "channel",
+    });
+    const { openDiscuss } = await start();
+    openDiscuss(channelId);
+    await click(".o-mail-Discuss-header button[title='Notification Settings']");
+    await click(".o-discuss-NotificationSettings button[title='Mute Channel']");
+    await click(".dropdown-item[title='For 15 minutes']");
+
+    await click(".o-mail-Discuss-header button[title='Notification Settings']");
+    await contains(".o-discuss-NotificationSettings span", { text: "Unmute Channel" });
+    await click(".o-discuss-NotificationSettings button", { text: "Unmute Channel" });
+
+    await click(".o-mail-Discuss-header button[title='Notification Settings']");
+    await contains(".o-discuss-NotificationSettings span", { text: "Unmute Channel" });
+});
```

Comment:
- same remark with empty lines: no empty newline, it gives a shape that gives wrong impression there are more tests than there are, no newline keeps LOCs of a single test cohesive as it should
- `.o-mail-Discuss-header/.o-discuss-NotificationSettings button` are useless noise. Test is much easier to read and maintain if you just keep grepping the `text`/title part.